### PR TITLE
Clean up some top level projects

### PR DESCRIPTION
**Pull request by jherico**
## Description

This makes some small tweaks to the CMake config to migrate some of the projects that are cluttering the top level into proper subfolders.  The glslangValidator and the cli11 projects both create things at the top level or in folders under the top level that should be moved into the "Third Party" folder in IDEs supporting hierarchies.  Additionally, some targets that have been created by `add_custom_target` which are useful for bundling, but which are useless when navigating projects in an IDE have been moved to "Cmake/CustomTargets" (the current offline shader compilation WIP branch generates a TON of these)

The only caveat here is that the CLI11 project is generated as an `interface_library` target type, which has a limited set of properties that can be set on it prior to CMake version 3.19, so the minimum required has to be bumped in order to resolve that.  The minimum required CMake version can be left alone if needed but that one specific change will have to be reverted in that case.

## General Checklist:

This only modifies some meta properties for the project targets in CMake and the resulting location of the projects inside a hierarchical build IDE such as MSVC.  It has no impact on compilation or linking properties, nor changes any sample code, so most of these checks are superfluous in this instance.

- [X] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [X] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [ ] ~I have commented any added functions (in line with Doxygen)~
- [ ] ~I have commented any code that could be hard to understand~
- [X] My changes do not add any new compiler warnings
- [ ] ~My changes do not add any new validation layer errors or warnings~
- [ ] ~I have used existing framework/helper functions where possible~
- [ ] ~My changes do not add any regressions~
- [ ] ~I have tested every sample to ensure everything runs correctly~
- [ ] ~This PR describes the scope and expected impact of the changes I am making~

 Note: The Samples CI runs a number of checks including:
 - [X] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [X] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)

### CODE REVIEW

File: third_party/CMakeLists.txt
Comment (SaschaWillems) [inline third_party/CMakeLists.txt]: We have CMake 3.12+ as min. requirements in our build instructions. We should decide if we really want to bump that version and if yes, we need to adjust our documentation accordingly.

Comment (jherico) [reply third_party/CMakeLists.txt]: The top level project currently has `cmake_minimum_required(VERSION 3.16)`, and I was actually surprised that I found `cmake_minimum_required` declarations all over the project, since it's almost always set once, at the top level and that's it.  

Trying to revert to the `3.12` requirement as documented would probably be pretty tricky, although I can understand if you want avoid updating beyond 3.16, since out of the box Ubuntu 20.04 only supports up to 3.16 I think.  I do understand that as maintainers of an example repository it's important for you to strike a balance between wanting the features and improvements of more recent CMake versions and needing to support people who are pinned on older versions for whatever reason (usually the Android NDK CMake version in my experience).

Like I said, the only specific change that requires this version bump is CLI11, so if it's decided that it's undesirable to make this change I can remove that one.  For now I'll look try to bring the docs in alignment with this value.

Comment (SaschaWillems) [reply third_party/CMakeLists.txt]: Yeah, it's kinda messy right now. I think we need to discuss this internally, as we have people compiling our samples on some very specific platforms. Not sure if those platforms do have access to recent CMake versions.

**Comment by SaschaWillems** [comment]
Thanks a lot for doing this. Having some of the projects at top-level in the project browser was always kinda annoying. Aside from the CMake bump, which we need to discus internally @marty-johnson59 this looks good.

**Comment by jherico** [comment]
I've reverted the minimum required declaration, but left the folder setting for now.  I believe what happens now is that if you're using CMake < 3.19 it emit a warning that you can't set the folder property on interface libraries, but doesn't prevent the CMake configuration from finishing.

**Review by gary-sweet — APPROVED** [review]
LGTM. CMake 3.19 would have been fine for me (since the question was asked on the samples call).

**Comment by marty-johnson59** [comment]
Thanks!

**Review by SaschaWillems — APPROVED** [review]

**Review by asuessenbach — APPROVED** [review]

**Comment by marty-johnson59** [comment]
Merging - 3 approvals

**Comment by SaschaWillems** [comment]
Looks like this breaks compatibility with older CMake versions. We currently state CMake 3.12 or newer in our build document. So we either need to fix the CMakeList or update our requirements (I prefer the latter).

**Comment by jherico** [comment]
I agree with Sascha that updating the minimum CMake version should be strongly considered, perhaps even going so far as to including instructions on how to bootstrap a private CMake version if you're stuck on something older.

While this is an example repository and should be aimed at the widest audience, it's essentially a top level project and doesn't have downstream consumers who might be constrained in their development environment.  While some example creators might be so constrained, I think it's reasonable that this should be prioritized lower relative to the ability to use more advanced CMake features.  

Related: https://www.scivision.dev/cmake-changelog/

**Pull request merged**
